service/handlers.go

```diff
@@ -143,6 +153,23 @@ func SignHandler(w http.ResponseWriter, r *http.Request) ErrorResponse {
 		return ErrorInvalidType
 	}
 
+	// Double check the model assertion if present
+	if modelAssert != nil {
+		if modelAssert.Type() != asserts.ModelType {
+			logMessage("SIGN", "invalid-type", "The 2nd assertion type must be 'model'")
```

Comment:
maybe service/errors.go can be populated with other error messages for this case. A kind of "ErrorInvalidSecondType" or whatever

Reply:
ok